src/main/java/org/mule/service/http/impl/service/server/grizzly/ExecutorPerServerAddressIOStrategy.java

```diff
@@ -69,17 +72,27 @@ public boolean executeIoEvent(final Connection connection, final IOEvent ioEvent
 
   @Override
   public Executor getThreadPoolFor(Connection connection, IOEvent ioEvent) {
-    if (WORKER_THREAD_EVENT_SET.contains(ioEvent)) {
+    if (useSelectorThread(connection, ioEvent)) {
+      return null;
+    } else {
       final InetSocketAddress socketAddress = (InetSocketAddress) connection.getLocalAddress();
       final InetAddress address = socketAddress.getAddress();
       final int port = socketAddress.getPort();
       return executorProvider.getExecutor(new DefaultServerAddress(address, port));
-    } else {
-      // Run other types of IOEvent in selector thread.
-      return null;
     }
   }
 
+  private boolean useSelectorThread(Connection connection, IOEvent ioEvent) {
```

Comment:
looking at who calls this and what it does internally, it seems to be a case of double negation. Can you rename this to `mustSwitchThread` and refactor accordingly?